### PR TITLE
Improve CSP report diagnostics

### PR DESCRIPTION
**Pull request by strickvl**
## Summary

CSP reports now identify the blocked resource without retaining query strings, fragments, embedded non-HTTP payloads, or malformed attacker-controlled values. This gives the next Pages-to-Workers cutover enough detail to distinguish known Pages noise from a new Worker regression.

This PR changes diagnostic logging only. It does not alter CSP enforcement or production routing.

## Changes

- Preserve safe CSP tokens such as `eval`.
- Reduce HTTP(S) values to their origin and query-free, fragment-free path.
- Preserve only allowlisted `data:` and `blob:` schemes, including their bare report forms.
- Map every other parsed custom scheme to a fixed marker.
- Apply the same privacy boundary to document and blocked URIs.
- Reject non-string values and use a fixed marker for malformed strings.
- Add focused regression coverage for privacy and diagnostic edge cases.

## Reviewer focus

Please check the sanitizer's privacy boundary and whether the safe CSP keyword allowlist is appropriately narrow.

## Validation

- `pnpm test` (146 tests)
- `pnpm check`
- `pnpm check:tests`
- `pnpm lint`
- `pnpm build`
- `pnpm smoke:dist`
- `pnpm check:worker` (18/18 runtime checks)

The implementation also received correctness, security, testing, project-standards, and independent adversarial review. All actionable findings were applied.

## Post-Deploy Monitoring & Validation

- Watch Worker or Pages logs for `[csp-report]`.
- Healthy: `blockedResource` contains only a safe CSP token, an HTTP(S) origin plus query-free path, `data:`, `blob:`, an empty missing value, `(other-scheme)`, or `(unparseable)`.
- Failure: query strings, fragments, embedded payload contents, custom scheme names, or array/object values appear in logs; malformed reports cause non-204 responses.
- Observe during the next staging smoke and production cutover window.
- Owner: Alex and Codex during the supervised migration.
- Mitigation: revert this commit if sensitive values appear. This PR does not change CSP enforcement or production routing.

Related: #169

### CODE REVIEW

File: src/pages/api/csp-report.ts [hunk 1]
```diff
@@ -10,21 +10,54 @@ import type { APIContext } from "astro";
...
+  if (typeof value !== "string" || value.length === 0) {
+    return "";
+  }
+
+  if (SAFE_CSP_KEYWORDS.has(value)) {
+    return value;
+  }
+
+  try {
+    const url = new URL(value);
```
[the remark this answers is not in the document]
Comment (strickvl) [reply src/pages/api/csp-report.ts:38]: > Recognize the expected bare scheme tokens before attempting URL parsing.

Addressed in `1b82738`: the sanitizer now normalizes bare `data` and `blob` values, retains only those allowlisted non-HTTP schemes, and maps every other parsed custom scheme to `(other-scheme)` so attacker-controlled scheme names cannot reach logs. Regression tests cover both bare values and a sentinel-bearing custom scheme.
